### ISSUE TITLE
Rows Modified Support

### ISSUE THREAD

**Issue by baderkha** [go-rel/rel#162]
Hey there , just wanted to say nice job on this package , it solves a lot of my concerns having a layered architecture . :) 

I just wanted to ask is there any way to get the RowsModified from a sql query , i know gorm has it as 

```
rowsAffected := db.
                            Where("field=?",someVal).
                            Where("otherField=?", somVal2).
                            Update(&record).
                            RowsAffected
```

**Comment by Fs02** [comment]
Hi, Thanks for taking a look at this package :)

hmm, interesting...
Currently, `RowsAffected` is not returned by `UpdateAll` function.
It's easy to make the [adjustments](https://github.com/go-rel/rel/blob/b13893bc8de47625c00c8acfdf5a4d1814ef886f/repository.go#L723), but requires breaking changes 🤦 

may I know what is the use case? is it to update multiple records only? I assume you won't need RowsAffected when updating only one record 🤔

**Comment by baderkha** [comment]
Hey , thanks for getting back. Yeah for 1 record it doesn't make sense but there are some use cases where you want to know the rows modified when applying a bulk update , or bulk insert operation . Forexample , 

if i have 2 tables and i wanted to copy data from table 1 to table 2 dynamically via a query

Insert from select statement ,then i would want to show the client how many records i wrote or updated

**Comment by Fs02** [comment]
> or bulk insert operation

do you also need number of inserted rows for `InsertAll`? it's currently doesn't make sense to return that in `InsertAll`, because the insertion is performed as a single query, so I think it should be all success or all failed 🤔

**Comment by baderkha** [comment]
you would need this for mutation queries where we wouldn't know the records being modified beforehand ie , Exec , UpdateAll 

for exec because i can do something like this ``` INSERT INTO someTable (field1,field2) Select field 1,field2 FROM someTable2  ```

**Comment by Fs02** [comment]
Opened pull request for affected rows support on UpdateAll and DeleteAll function here https://github.com/go-rel/rel/pull/164

> I see that the exec feature had not been merged

Yeah, I'll take a look again on the PR later

**Comment by baderkha** [comment]
Thank you so much @Fs02 !!! you rock

**Comment by Fs02** [comment]
released on v0.14.0, thanks for your suggestion, please feel free to open a new issue if you have more question
